lxd/task/task.go

```diff
@@ -49,7 +50,7 @@ func (t *Task) loop(ctx context.Context) {
 			// never expire (hence the task function won't ever be
 			// run, unless Reset() is called and schedule() starts
 			// returning values greater than zero).
-			if schedule > 0 {
+			if schedule > 0 || t.cron {
```

Comment:
We need to know that it's a cron schedule as we need to do `delay, _ = t.schedule()` instead of ` delay = schedule` in that case since the schedule won't always be the same.

Reply:
But the ```schedule``` in ```delay = schedule``` comes from ```schedule, err := t.schedule()```. I'm confused since the whole point of the Schedule type is to return schedule intervals that aren't necessarily always the same. Let's move the conversation on IRC if it helps.